tools/pipelines/templates/include-publish-npm-package.yml

```diff
@@ -22,6 +26,15 @@ stages:
       feedName: https://pkgs.dev.azure.com/fluidframework/internal/_packaging/test/npm/registry/
       environment: test-package-build-feed
       official: false
+      artifact: pack-scoped
+  - ${{ if eq(parameters.publishNonScopedPackages, true) }}:  
+    - template: include-publish-npm-package-steps.yml
+      parameters:
+        namespace: false
+        feedName: https://pkgs.dev.azure.com/fluidframework/internal/_packaging/test/npm/registry/
+        environment: test-package-build-feed
+        official: false
+        artifact: pack-non-scoped
```

Comment:
nit:  This would spread the publish to two machines, which may be a little bit of a waste here.  It would be better if they are templated steps instead of templated job for the scope/non-scope.  

Also, this reminded me that we do have a slight problem with our publishing, where we don't do it in a topological order based on dependencies.   So there is a period of time where people install our package would fail, because we have only partially published our packages.  We will need to able to control order to fix it, and will probably be good to do it on a single job. 

This is probably ok for now, but we should fix the publish order before we release 1.0.

Reply:
I looked into updating the jobs here to be templated steps instead. However, the deployment keyword is only available as a job. To do this, we need to move the deployment up to `include-publish-npm-package.yml` where the job is declared and then use the template just for the steps. I'll set up a follow-up PR for this since it involves modifying some of the other parameters as well.

As for publishing order, I'm guessing this order would be similar to how our repo is ordering the packages to build when running `npm run build:fast`?